connect/file/src/main/java/org/apache/kafka/connect/file/FileStreamSourceConnector.java

```diff
@@ -56,20 +58,12 @@ public String version() {
 
     @Override
     public void start(Map<String, String> props) {
-        filename = props.get(FILE_CONFIG);
-        topic = props.get(TOPIC_CONFIG);
-        if (topic == null || topic.isEmpty())
-            throw new ConnectException("FileStreamSourceConnector configuration must include 'topic' setting");
+        AbstractConfig parsedConfig = new AbstractConfig(CONFIG_DEF, props);
+        filename = parsedConfig.getString(FILE_CONFIG);
+        topic = parsedConfig.getString(TOPIC_CONFIG);
```

Comment:
Is this sufficient to rule out null/empty topic names? Should we have a test case?

Reply:
Indeed, the config string property returns non-empty and trimmed. 

Good point about the test case. I saw it in the task's tests and thought we had it. But the task gets the parsed config, so this test is a no-op now (it can't happen). Thus, I moved the test to the connector tests that does the parsing and then passes the parsed property map to the task. (also removed the check for `null` in the task, because that can't happen after we've parsed the config too).

Reply:
Can you clarify where the non-null/trim checks occur? Is it in the `AbstractConfig` parsing above or does it happen before the connector is started.

Reply:
The reason I ask is that I was expecting this would be implemented with a `ConfigDef` validator, but I didn't see one.

Reply:
This basic functionality is baked in `ConfigDef`. 
Trimming happens here: https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java#L641-L642
and then check for `null` and class type here:
https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java#L665-L668

I believe a validator would be required if the config required something more than this basic check for non-null and trimming. 

Given that, I've been surprised to see changes like this one: https://github.com/apache/kafka/pull/4525/files

Apparently this functionality is a bit hidden. But since it's there I believe it should be promoted to avoid duplicate checks and code.

Reply:
Hmm, but if the value is null, we won't hit those points: https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java#L638. With this patch, both of these tests fail:

```java
    @Test(expected = ConfigException.class)
    public void testEmptyTopicNotAllowed() {
        sourceProperties.put(FileStreamSourceConnector.TOPIC_CONFIG, "");
        connector.start(sourceProperties);
    }

    @Test(expected = ConfigException.class)
    public void testNullTopicNotAllowed() {
        sourceProperties.put(FileStreamSourceConnector.TOPIC_CONFIG, null);
        connector.start(sourceProperties);
    }
```

Reply:
Regarding `testNullTopicNotAllowed`, there's no way to pass `null` by parsing a properties file. 

Regarding `testEmptyTopicNotAllowed` I left it on purpose, because somehow the intention to use an empty string as a topic name is explicit. But I can see how this can lead to issues when rendering property templates in tests etc, so I'll bring back the check for empty.

Reply:
Actually, to keep things succinct, I'm looking at the "future" where such property could be a list. 

I'm making this config a list and apply a check at the same time for empty, or multiple topics, allowing only a single topic to be defined. Added a case for blank 'topic' property though per your suggestion. 

Thanks for mentioning!